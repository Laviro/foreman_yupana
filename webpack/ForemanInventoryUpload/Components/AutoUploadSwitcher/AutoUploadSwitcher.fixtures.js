export const currentAutoUploadEnabled = false;

export const handleToggleResponse = { data: { autoUploadEnabled: true } };
