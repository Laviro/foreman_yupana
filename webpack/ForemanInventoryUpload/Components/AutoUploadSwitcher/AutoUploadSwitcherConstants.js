export const AUTO_UPLOAD_TOGGLE = 'AUTO_UPLOAD_TOGGLE';
