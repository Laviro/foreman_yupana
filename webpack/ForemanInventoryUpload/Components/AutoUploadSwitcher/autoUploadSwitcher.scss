.auto_upload_switcher {
  margin-bottom: 5px;
}
