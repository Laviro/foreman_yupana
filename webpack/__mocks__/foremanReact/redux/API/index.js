export const get = data => ({ type: 'get-some-type', ...data });
export const put = data => ({ type: 'put-some-type', ...data });
export const post = data => ({ type: 'post-some-type', ...data });
export const patch = data => ({ type: 'patch-some-type', ...data });

export const API = {
  get: jest.fn(),
  put: jest.fn(),
  post: jest.fn(),
  delete: jest.fn(),
  patch: jest.fn(),
};
