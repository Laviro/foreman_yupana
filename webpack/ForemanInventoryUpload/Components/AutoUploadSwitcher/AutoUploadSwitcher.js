import React from 'react';
import PropTypes from 'prop-types';
import { Switch, FieldLevelHelp } from 'patternfly-react';
import { translate as __ } from 'foremanReact/common/I18n';
import './autoUploadSwitcher.scss';

const AutoUploadSwitcher = ({ autoUploadEnabled, handleToggle }) => (
  <div className="auto_upload_switcher">
    <span>Auto upload</span>
    <FieldLevelHelp
      content={__(
        'Enable automatic upload of your host inventory to the Red Hat cloud'
      )}
    />
    <Switch
      size="mini"
      value={autoUploadEnabled}
      onChange={() => handleToggle(autoUploadEnabled)}
    />
  </div>
);

AutoUploadSwitcher.propTypes = {
  autoUploadEnabled: PropTypes.bool,
  handleToggle: PropTypes.func.isRequired,
};

AutoUploadSwitcher.defaultProps = {
  autoUploadEnabled: true,
};

export default AutoUploadSwitcher;
