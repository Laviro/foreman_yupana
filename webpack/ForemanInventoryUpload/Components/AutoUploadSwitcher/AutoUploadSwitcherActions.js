import API from 'foremanReact/API';
import {
  AUTO_UPLOAD_TOGGLE,
  AUTO_UPLOAD_TOGGLE_ERROR,
} from './AutoUploadSwitcherConstants';

export const handleToggle = () => async dispatch => {
  try {
    const {
      data: { isAutoUpload },
    } = await API.post('toggle_auto_upload');
    dispatch({
      type: AUTO_UPLOAD_TOGGLE,
      payload: {
        isAutoUpload,
      },
    });
  } catch (error) {
    dispatch({
      type: AUTO_UPLOAD_TOGGLE_ERROR,
      payload: {
        error: error.message,
      },
    });
  }
};
