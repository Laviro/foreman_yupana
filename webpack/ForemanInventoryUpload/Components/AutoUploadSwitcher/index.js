import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as actions from './AutoUploadSwitcherActions';
import AutoUploadSwitcher from './AutoUploadSwitcher';
import { selectIsAutoUpload } from '../AccountList/AccountListSelectors';

// map state to props
const mapStateToProps = state => ({
  isAutoUpload: selectIsAutoUpload(state),
});

// map action dispatchers to props
const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

// export connected component
export default connect(mapStateToProps, mapDispatchToProps)(AutoUploadSwitcher);
