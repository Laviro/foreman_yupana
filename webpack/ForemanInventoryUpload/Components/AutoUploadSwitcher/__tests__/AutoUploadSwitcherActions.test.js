import { testActionSnapshotWithFixtures } from 'react-redux-test-utils';
import API from 'foremanReact/API';
import { handleToggle } from '../AutoUploadSwitcherActions';
import { handleToggleResponse } from '../AutoUploadSwitcher.fixtures';

jest.mock('foremanReact/API');
API.post.mockImplementation(async () => handleToggleResponse);

const fixtures = {
  'should handleToggle': () => handleToggle(),
};

describe('AutoUploadSwitcher actions', () =>
  testActionSnapshotWithFixtures(fixtures));
