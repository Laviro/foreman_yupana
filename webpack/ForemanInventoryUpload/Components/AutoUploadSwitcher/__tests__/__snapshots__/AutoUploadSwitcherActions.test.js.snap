// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoUploadSwitcher actions should handleToggle 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "autoUploadEnabled": true,
      },
      "type": "AUTO_UPLOAD_TOGGLE",
    },
  ],
]
`;

exports[`AutoUploadSwitcher actions should handleToggle with error 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "message": Object {
          "message": "Network error!",
          "sticky": true,
          "type": "error",
        },
      },
      "type": "TOASTS_ADD",
    },
  ],
]
`;
